src/Fields/FieldsContract.php

```diff
@@ -165,21 +166,29 @@ public function getCollectionsValuesForResponse(Field $field, FieldResponse $res
 
         $response = Arr::wrap($response);
 
-        // to not braking old dataSource structure
-        if ((int) $field->options['dataSource'] !== 0) {
-            $response = BoltPlugin::getModel('Collection')::query()
-                ->find($field->options['dataSource'])
-                ?->values
-                ->whereIn('itemKey', $response)
-                ->pluck('itemValue')
-                ->join(', ') ?? '';
-        } else {
+        $dataSource = (int) $field->options['dataSource'] ?? $field->options['dataSource'];
+        $cacheKey = 'dataSource_' . $dataSource . '_response_' . md5(serialize($response));
```

Comment:
what if we want to fluch a chave for spsefic collection? for example after edit any collection its good idea to flush the cache for that record, but this way we cant! ince there is no response id.

I guess we can do something like `collection.$id.$responseid.mdf`..

Reply:
so this way we can flush all cached keys for collection id:1

Reply:
hmm , 

But , i think is not work well , because all response value will be back from first result , i mean  if one submitted record has two collection field with same id  but values($response) different ,  the results in table will be show wrong 

am i right ?

Reply:
yea, did some testing and no success, will keep this and see how it goes! :)

I updated line 169 since `phpstan: Expression on left side of ?? is not nullable.`
and some improvements.

thank you again, do last test if you can and let me know to merge :)